runtime/src/prioritization_fee_cache.rs

```diff
@@ -121,20 +109,26 @@ impl PrioritizationFeeCacheMetrics {
 enum CacheServiceUpdate {
     TransactionUpdate {
         slot: Slot,
+        bank_id: BankId,
         transaction_fee: u64,
         writable_accounts: Arc<Vec<Pubkey>>,
     },
     BankFrozen {
         slot: Slot,
+        bank_id: BankId,
     },
     Exit,
```

Comment:
hmm, how about adding new `BankDumped { slot: Slot }`, instead of introducing this `BankId`-based 2nd level caching? And the variant can be sent inside `purge_unfirmed_duplicate_slot()` after plumbing `&PriotizationFeeCache` there.

that alternative impl is simpler and should be safe as long as `solPrFeeCacSvc` is single threaded, which should be because of already existing linearizable  requirement of `BankFrozen`.

Reply:
Agreed. But then I thought that if I had missed `purge_unconfirmed_duplicated_slot()` last time, what would make sure I won't miss something else, or new duplicated slot/bank logic being added without being aware of `prioritization_fee_cache`? That motivates to explore the approach of purge cache by OC event, which should be definitive and un-rollbackable. But I do agree it adds some complicity, and using `bank_id` as secondary key may not be ideal. Hence this is a draft. 😃  Keen to hear your thoughts on it.

Reply:
yeah, i see your reasoning for bank_id approach clearly.

i spent some time on which is better. and i couldn't conclude it. so, it boils down to the author's preference.

as said bank_id takes more human effort than `BankDumped` approach. yet, i saw you added tests. seems you're well energized for the effort. :)

please continue. once this pr is out of draft, i'll review.

Reply:
Totally agree that this approach is bit more involved, but not too bad :) I reasoned to myself that it is worth the extra effort to make p.fee cache bit more independent.

I added test thinking either approach we will need a test to capture the corner case scenario. 

I'd like to add a metric counter to capture the occurrence of dup bank, then will move it out of draft.

Thanks for being patient ❤️